tools/taskcluster-worker-runner/provider/provider/common.go

```diff
@@ -40,3 +46,19 @@ func RegisterWorker(state *run.State, wm tc.WorkerManager, workerPoolID, provide
 
 	return nil
 }
+
+// RemoveWorker will request worker-manager to terminate the given worker, if it
+// fails, it shuts down us
+func RemoveWorker(wc tc.WorkerManager, wi *WorkerInfo) error {
+	err := wc.RemoveWorker(wi.WorkerPoolID, wi.WorkerGroup, wi.WorkerID)
+	if err != nil {
+		log.Printf("Error removing the worker: %v\n", err)
+		log.Printf("Falling back to system shutdown")
+		if err = Shutdown(); err != nil {
+			log.Printf("Error shutting down the worker: %v\n", err)
+			return err
+		}
+	}
```

Comment:
Would it make sense to always do both?

Reply:
Initially, I thought about that. But in the current state, if we see any instance in the "stopped", we know something went wrong with `RemoveWorker`, and can dig into the logs to investigate it.

Reply:
Makes sense!